### PR TITLE
chore(warehouse): tracking plan validation compatiblility for spread sheet plugin

### PR DESCRIPTION
**Pull request by achettyiitr**
> Tracking plan compatibility for spreadsheet plugin. https://www.notion.so/rudderstacks/Tracking-plan-compatibility-for-spreadsheet-plugin-4754cc2b3fc64038829a8088e7f8e72d

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [x] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [x] Changes have been reviewed by at least one other engineer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: util/eventValidation.js [hunk 1]
```diff
@@ -277,7 +276,8 @@ async function handleValidation(event) {
         }
 
         const violationsByType = new Set(validationErrors.map(err => err.type));
-        for (const [key, value] of Object.entries(mergedTpConfig)) {
+        for (const [key, val] of Object.entries(mergedTpConfig)) {
+            const value = val.toString().toLowerCase();
```
Comment (Jayachand) [inline util/eventValidation.js:280]: use `val.?toString()?.toLowerCase()` to handle null values too

**Review by Jayachand — APPROVED** [review]

**Review by satishrudderstack — APPROVED** [review]
LGTM

File: util/eventValidation.js [hunk 2]
```diff
@@ -277,7 +276,8 @@ async function handleValidation(event) {
         }
 
         const violationsByType = new Set(validationErrors.map(err => err.type));
-        for (const [key, value] of Object.entries(mergedTpConfig)) {
+        for (const [key, val] of Object.entries(mergedTpConfig)) {
+            const value = val?.toString()?.toLowerCase();
```
Comment (Jayachand) [inline util/eventValidation.js:280]: Currently spreadsheet gives boolean `true/false` while config-backend api is configured to string `"true"/"false"`. Minor suggestion: what do you think of using boolean everywhere? not right away, some other time

**Review by Jayachand — DISMISSED** [review]
LGTM

**Review by psrikanth88 — DISMISSED** [review]

**Review by satishrudderstack — DISMISSED** [review]
LGTM

**Review by satishrudderstack — APPROVED** [review]
LGTM

**Review by sivashanmukh — APPROVED** [review]
LGTM

**Review by Jayachand — APPROVED** [review]

**Pull request merged**